src/string_buffer.rs

```diff
@@ -172,6 +170,14 @@ impl StringNode {
         // Otherwise, try to find a newline in the current node.
         result.or_else(|| self.data.rfind('\n').map(|i| self.total_len() - i - 1))
     }
+
+    // Returns a flat clone of a node
```

Comment:
It's supposed to clone the node in such a way that it is completely independent/standalone (i.e. without links and such). I'll add it to the comment.